src/MessageTrait.php

```diff
@@ -66,7 +66,7 @@ public function getProtocolVersion()
      * new protocol version.
      *
      * @param string $version HTTP protocol version
-     * @return self
+     * @return static
```

Comment:
@Ocramius The issue here is the inheritance tree, which includes interfaces, and, more specifically, traits. Per [this comment from @mvriel](https://github.com/zendframework/zend-diactoros/pull/37#issuecomment-107195929), `self` indicates that it's an instance of the class where the docblock is defined, while `static` is defined to return an instance of the _current_ class, following inheritance.

So, a change is required here for sure. The question is whether changes are needed on the PSR-7 interfaces. I'd argue they are for `MessageInterface` and `RequestInterface`, as those two both are inherited by other interfaces in the specification.